### PR TITLE
Adrubocop

### PR DESCRIPTION
**Pull request by apalski**
This PR is for the installation and setup of Rubocop for checking app compliance with the Ruby Style Guide:
- Installed Rubocop, added gem to gemfile and created enable.yml, disable.yml and default.yml.
- Changed setting 'for magic comments' in enabled.yml.
Also, removed unused gems and their dependents while adding the Rubocop gem to the gemfile.
I'm hoping for feedback on the Rubocop setup and included files.
@elle @scara82 @jonathanly @MelissaKaulfuss

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -7,19 +7,17 @@ end
...
-gem "flutie"
 gem "honeybadger"
 gem "jquery-rails"
+gem "omniauth"
+gem "omniauth-facebook"
 gem "pg"
 gem "puma"
-gem "rack-canonical-host"
 gem "rails", "~> 5.1.3"
-gem "recipient_interceptor"
```
Comment (elle) [inline Gemfile]: That's a good gem to have... but up to you...

File: Gemfile [hunk 2]
```diff
@@ -38,15 +36,16 @@ group :development, :test do
   gem "awesome_print"
   gem "bullet"
   gem "bundler-audit", ">= 0.5.0", require: false
+	gem "capybara-webkit"
```
Comment (elle) [inline Gemfile]: Indentation??

File: Gemfile [hunk 3]
```diff
@@ -38,15 +36,16 @@ group :development, :test do
...
   gem "dotenv-rails"
   gem "factory_bot_rails"
   gem "pry-byebug"
   gem "pry-rails"
-  gem "rspec-rails", "~> 3.6"
+  gem "rspec-rails", "~> 3.7"
+	gem "rubocop"
```
Comment (elle) [inline Gemfile]: Indentation?

File: Gemfile [hunk 4]
```diff
@@ -38,15 +36,16 @@ group :development, :test do
   gem "awesome_print"
   gem "bullet"
   gem "bundler-audit", ">= 0.5.0", require: false
+	gem "capybara-webkit"
   gem "dotenv-rails"
   gem "factory_bot_rails"
   gem "pry-byebug"
   gem "pry-rails"
```
Comment (elle) [inline Gemfile]: I would probably just keep "pry-byebug".
So line 43 is redundant

**Comment by elle** [comment]
Also, this PR is adding oath gems that should belong in a PR about authentication :)

**Comment by elle** [comment]
Sidenote: example `.rubucop.yml` https://github.com/elle/gilded-rose/blob/master/.rubocop.yml

File: Gemfile (hunk 1 above)
Comment (apalski) [reply Gemfile]: Thanks for the feedback Elle, I will create .rubocop.yml and use it for the overrides. I thought maybe I'd keep enable and disable as a guide to what things I can change but I guess I can get rid of them and look them up as needed.

File: .gitignore [hunk 5]
```diff
@@ -10,6 +10,9 @@
 /test/version_tmp/
 /tmp/
 
+/log/bullet.log
+/log/development.log
```
Comment (elle) [inline .gitignore]: This should probably change to be more dynamic, like: `/log/*.log`

**Comment by elle** [comment]
And the general rule is to not keep code that is not used around. You could always look up the appropriate rules for rubucop.

**Comment by elle** [comment]
Should have mentioned this before. The `default.yml` doesn't need to be here as well

**Comment by elle** [comment]
LGTM

**Pull request closed without merging**